From 6270425f212a335336a7b68a43c729f79f3914d8 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Fri, 2 Nov 2018 10:53:30 +0200
Subject: [PATCH 5057/5242] MLK-20181-9: arch64: dts: fsl-imx8dx: Add axi clk
 to adma_lcdif

commit  07a423d3beb1cf31f08ce670b6e854468bc74416 from
https://source.codeaurora.org/external/imx/linux-imx.git

Currently, the adma_lcdif clocks are wrong. Correct these clocks.
Fix suggestion received from:
Ranjani Vaidyanathan <ranjani.vaidyanathan@nxp.com>

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Acked-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi |   16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index 484e812..09618ac 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -858,7 +858,6 @@
 					power-domains = <&pd_dma_elcdif_pll>;
 				};
 			};
-
 			pd_dma_flexcan0: PD_DMA_CAN_0 {
 				reg = <SC_R_CAN_0>;
 				#power-domain-cells = <0>;
@@ -1774,11 +1773,16 @@
 	adma_lcdif: lcdif@5a180000 {
 		compatible = "fsl,imx8qxp-lcdif", "fsl,imx28-lcdif";
 		reg = <0x0 0x5a180000 0x0 0x10000>;
-		clocks = <&clk IMX8QXP_LCD_CLK>, <&clk IMX8QXP_LCD_IPG_CLK>;
-		clock-names = "pix", "disp_axi";
-		assigned-clocks = <&clk IMX8QXP_LCD_SEL>, <&clk IMX8QXP_ELCDIF_PLL>;
-		assigned-clock-rates = <804000000>, <804000000>;
-		assigned-clock-parents = <&clk IMX8QXP_ELCDIF_PLL>;
+		clocks = <&clk IMX8QXP_LCD_CLK>,
+			 <&clk IMX8QXP_LCD_PXL_CLK>,
+			 <&clk IMX8QXP_LCD_IPG_CLK>;
+		clock-names = "pix", "disp_axi", "axi";
+		assigned-clocks = <&clk IMX8QXP_LCD_SEL>,
+				  <&clk IMX8QXP_LCD_PXL_SEL>,
+				  <&clk IMX8QXP_ELCDIF_PLL_DIV>;
+		assigned-clock-parents = <&clk IMX8QXP_ELCDIF_PLL>,
+					 <&clk IMX8QXP_LCD_PXL_BYPASS_DIV>;
+		assigned-clock-rates = <0>, <24000000>, <804000000>;
 		interrupts = <GIC_SPI 62 IRQ_TYPE_LEVEL_HIGH>;
 		power-domains = <&pd_dma_lcd0>;
 		status = "disabled";
-- 
1.7.9.5

